torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -267,7 +267,7 @@ void FunctionParameter::set_default_str(const std::string& str) {
       default_intlist = parse_intlist_args(str, size);
     }
   } else if (type_ == ParameterType::SCALARTYPE) {
-    if (str == "None") {
+    if (str == "None" || str == "c10::nullopt") {
```

Comment:
why here is `c10::nullopt`? python arg parser should only consume `None` in python rather than `nullopt`

Reply:
hmm... I hadn't thought about it in detail since this did fix an error I was getting at this point in the code. The default value is set in c++ though right? If the C++ function is defined like `Tensor::mean(Tensor self, c10::optional<ScalarType> dtype = c10::nullopt)`, and the python function call is tensor.mean(), where should the parser find the conversion from `c10::nullopt` to `None`?